hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/sequences/AlignedReadPair.java

```diff
@@ -92,4 +108,151 @@ private void divideRecords(List<SAMRecord> records)
+
+		/**
+		 * One {@link AlignedRead} mapped multiple times and the other only had supplementary or secondary alignments.
+		 */
+		ONE_MULTIMAPPED_ONE_MULTIMAPPED_SUPPLEMENTARY_ONLY(AlignedRead.Type.MULTIMAPPED,
+				AlignedRead.Type.MULTIMAPPED_SUPPLEMENTARY_ONLY),
+
+		/**
+		 * The data is deemed invalid. Note that if even one {@link AlignedRead.Type} is invalid, the
+		 * {@link AlignedReadPair.Type} should be deemed as {@link #INVALID}.
+		 */
+		INVALID(AlignedRead.Type.INVALID, AlignedRead.Type.INVALID);
+
+		/**
+		 * Stores an {@link AlignedRead.Type}{@code s} that together with {@code secondAlignedReadType} represents a
+		 * specific {@link AlignedReadPair.Type}.
+		 */
+		private AlignedRead.Type firstAlignedReadType;
+
+		/**
+		 * Stores a {@link AlignedRead.Type}{@code s} that together with {@code firstAlignedReadType} represents a
+		 * specific {@link AlignedReadPair.Type}.
+		 */
+		private AlignedRead.Type secondAlignedReadType;
+
+		private AlignedRead.Type getFirstAlignedReadType()
+		{
+			return firstAlignedReadType;
+		}
+
+		private AlignedRead.Type getSecondAlignedReadType()
+		{
+			return secondAlignedReadType;
+		}
+
+		/**
+		 * Constructor for AlignedReadPair.Type.
+		 * 
+		 * @param firstAlignedReadType
+		 *            {@link AlignedRead.Type} One of the two {@link AlignedRead.Type}{@code s} that define a specific
+		 *            {@link AlignedReadPair.Type}.
+		 * @param secondAlignedReadType
+		 *            {@link AlignedRead.Type} One of the two {@link AlignedRead.Type}{@code s} that define a specific
+		 *            {@link AlignedReadPair.Type}.
+		 */
+		private Type(AlignedRead.Type firstAlignedReadType, AlignedRead.Type secondAlignedReadType)
+		{
+			this.firstAlignedReadType = firstAlignedReadType;
+			this.secondAlignedReadType = secondAlignedReadType;
+		}
+
+		/**
+		 * Determines the {@link AlignedReadPair.Type} using two {@link AlignedRead.Type}{@code s}. If either
+		 * {@link AlignedRead.Type} is {@link AlignedRead.Type#INVALID}, returns {@link AlignedReadPair.Type#INVALID}.
+		 * Otherwise, returns one of the other possible types.
+		 * 
+		 * @param first
+		 *            {@link AlignedRead.Type}
+		 * @param second
+		 *            {@link AlignedRead.Type}
+		 * @return {@link AlignedReadPairType}
+		 */
+		private static Type determineType(AlignedRead.Type first, AlignedRead.Type second)
+		{
+			// Goes through all possible options (excluding if one is invalid while another is not).
+			for (Type alignedReadPairType : Type.values())
+			{
+				// As only the combination of the two AlignedReadTypes matter, checks both possibilities.
+				if (alignedReadPairType.getFirstAlignedReadType().equals(first)
+						&& alignedReadPairType.getSecondAlignedReadType().equals(second))
+					return alignedReadPairType;
+				else if (alignedReadPairType.getFirstAlignedReadType().equals(second)
+						&& alignedReadPairType.getSecondAlignedReadType().equals(first))
+					return alignedReadPairType;
+			}
```

Comment:
It's called for [every group of records from a single aligned read pair](https://github.com/molgenis/hadoop-pipeline/pull/8/files#diff-26c84876bb41c1ab84c21f42c39549b2R159), so a new implementation will be created and pushed.